shared/core/types/proto/actor.proto

```diff
@@ -12,12 +12,19 @@ enum ActorType {
   ACTOR_TYPE_VAL = 4;
 }
 
+enum StakeStatus {
+  UnknownStatus = 0;
+  Unstaking = 1;
+  Staked = 2;
+  Unstaked = 3;
+}
+
 message Actor {
   ActorType actor_type = 1;
   string address = 2;
   string public_key = 3;
-  repeated string chains = 4;
-  string generic_param = 5; // TODO(#313): Generalize `GenericParam` to a different name
+  repeated string chains = 4; // Not applicable to `Validator` actors
+  string service_url = 5; // Not applicable to `Application` actors
```

Comment:
I think it's a bit smelly that we have fields that are only applicable to some actors in here.

Especially now that `generic_param` was promoted to `service_url`

Not actionable for this PR of course.

I would perhaps close or modify #313 and leave a TODO here. Wdyt? I can help if needed